Updated Java driver to version 7.16.0

rashtao opened the pull request.

arcanefoam commented:
So basically Arango is now using Jackson ? The changes look really good and some of the other fixes are welcome. Thanks for taking your time for this.

Did you run the Tinkerpop tests?

rashtao commented:
Hi @arcanefoam 

Tests are passing, I have added GH actions, see https://github.com/rashtao/arangodb-tinkerpop-provider/actions/runs/12997958700

Nonetheless, some tests were failing (failing also with the older ArangoDB driver), so for the moment I have opted them out, see https://github.com/rashtao/arangodb-tinkerpop-provider/blob/6e81515f2b1745a62a2fc8ec8ed0b9e3a08ef0cc/src/main/java/com/arangodb/tinkerpop/gremlin/structure/ArangoDBGraph.java#L152-L164

rashtao commented:
@arcanefoam 
ArangoDB Java driver version 7 uses Jackson for serialization and deserialization of user data, see https://docs.arangodb.com/stable/develop/drivers/java/reference-version-7/serialization/#jacksonserde-default-user-data-serde

rashtao commented:
@arcanefoam 

my work here is done, CI is green: https://github.com/rashtao/arangodb-tinkerpop-provider/actions/runs/13028390417

Note that I have removed the filter tests (https://github.com/ArangoDB-Community/arangodb-tinkerpop-provider/pull/79/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L208-L213) so now all the test suites included in `ArangoDBTestSuite` are executed.

Many tests have been trivially fixed, i.e. customizing `ArangoDBGraphProvider` with specific vertex and edge collections, see https://github.com/ArangoDB-Community/arangodb-tinkerpop-provider/pull/79/files#diff-b63e847ab4ed732e2f079bfba0979318fe129214912bcad1457c223b5f197bb2

Tests that are still failing have been opted-out in `ArangoDBGraph`, see https://github.com/ArangoDB-Community/arangodb-tinkerpop-provider/pull/79/files#diff-a41d624f04d7e7c6ba0b8e0fa229b67c616c9a4b6bae734b105e623f0a95e4c8

In future, these could be reviewed, fixed or rewritten in a compatible way.

arcanefoam commented:
Thanks a lot for taking the time to do all the fixes. I am doing the review atm and all looks good.

arcanefoam commented on src/test/resources/logback-test.xml:
Why did you remove the appenders from the logger?

rashtao replied:
Sorry I have overlooked it...
Reverted in https://github.com/ArangoDB-Community/arangodb-tinkerpop-provider/pull/79/commits/19d86569ee6cbc54e8f0990e0444daa2b748850c
